addons/event/controllers/main.py

```diff
@@ -23,3 +26,63 @@ def event_ics_file(self, event, **kwargs):
+        :param event_id: the id of the prompted event. Only its attendees will be considered.
+        :param registration_ids: ids of event.registrations of which tickets are generated
+        :param badge_mode: boolean, True to use template of foldable badge instead of full page ticket.
+        :param simplified: boolean, True if we want to see the a simplified ticket in a size-adaptative html page.
+        """
+        event_id = int(event_id) if event_id else False
+        registration_ids = json.loads(registration_ids or '[]')
+        if not event_id or not dl_hash or not registration_ids:
+            raise Forbidden(_("These parameters to access tickets are invalid."))
+
+        event = request.env['event.event'].browse(event_id).exists()
+        hash_truth = event._get_tickets_dl_access_hash(registration_ids)
+        if not consteq(dl_hash, hash_truth):
+            raise Forbidden(_("These parameters to access tickets are invalid."))
+
+        event_registrations_sudo = self._fetch_registrations_for_event_tickets_dl(event_id, registration_ids)
+        if not event_registrations_sudo:
+            raise NotFound(_("Those registrations do not exist or are invalid."))
+
+        report_name_prefix = "Ticket" if simplified else "Badges" if badge_mode else "Tickets"
+        report_name = f"{report_name_prefix} - {event.name} ({event.date_begin_located})"
+        if len(event_registrations_sudo) == 1:
+            report_name += f" - {event_registrations_sudo[0].name}"
+
+        # sudo is necessary for accesses in template.
+        if simplified:
+            html = request.env['ir.actions.report'].sudo()._render_qweb_html(
+                'event.action_report_event_registration_simplified_ticket',
+                event_registrations_sudo.ids
+            )[0]
+            return request.make_response(html)
+        else:
+            pdf = request.env['ir.actions.report'].sudo()._render_qweb_pdf(
+                'event.action_report_event_registration_foldable_badge' if badge_mode else
+                'event.action_report_event_registration_full_page_ticket',
+                event_registrations_sudo.ids
+            )[0]
+            pdfhttpheaders = [
+                ('Content-Type', 'application/pdf'),
+                ('Content-Length', len(pdf)),
+                ('Content-Disposition', content_disposition(f'{report_name}.pdf'))
+            ]
+            return request.make_response(pdf, headers=pdfhttpheaders)
+
+    def _fetch_registrations_for_event_tickets_dl(self, event_id, registration_ids):
+        """ Fetch attendees in registration_ids belonging to event_id. Meant to allow override in order
+        to filter out registrations base on additional conditions. Helper method of /event/tickets_dl. """
+        return request.env['event.registration'].sudo().search([
+            ('id', 'in', registration_ids),
+            ('event_id', '=', event_id)
```

Comment:
(picky, for better git history in case osmeone add a new condition)
```suggestion
            ('event_id', '=', event_id),
```